packages/core/src/common/keybinding.ts

```diff
@@ -144,27 +151,62 @@ export class KeybindingRegistry {
         }
     }
 
-    registerKeybindings(...bindings: Keybinding[]): void {
+    registerKeybindings(...bindings: (RawKeybinding | Keybinding)[]): void {
         for (const binding of bindings) {
             this.registerKeybinding(binding);
         }
     }
 
+    protected keybindingFromRaw(binding: RawKeybinding): Keybinding | undefined {
+        if (this.commandRegistry.getCommand(binding.command)) {
+            const code = KeyCode.parse(binding.keybinding);
+            if (code) {
+                let context: KeybindingContext | undefined;
+                if (binding.context) {
+                    context = this.contextRegistry.getContext(binding.context);
+                }
+                return {
+                    commandId: binding.command,
+                    keyCode: code,
+                    contextId: context ? context.id : undefined,
+                    accelerator: binding.accelerator
+                };
+            } else {
+                this.logger.error(`Can't parse keybinding ${JSON.stringify(binding)}`);
+                return undefined;
+            }
+        }
+    }
+
     /**
      * Register a default keybinding to the registry.
      *
      * @param binding
      */
-    registerKeybinding(binding: Keybinding) {
+    registerKeybinding(inputBinding: Keybinding | RawKeybinding) {
+
+        let binding: Keybinding | undefined;
+        if (RawKeybinding.isRawKeybinding(inputBinding)) {
+            binding = this.keybindingFromRaw(inputBinding);
+        } else {
+            binding = inputBinding;
+        }
+
+        if (binding === undefined) {
+            return;
+        }
+
         const existingBindings = this.getKeybindingsForKeyCode(binding.keyCode);
         if (existingBindings.length > 0) {
-            const collided = existingBindings.filter(b => b.contextId === binding.contextId);
+            const collided = existingBindings.filter(b => binding !== undefined
```

Comment:
how can `undefined` end up in the array?

Reply:
That's a good question, to me it semms like it can't happen but tsc kept complaining.

This will go away anyway as I refactor parse to throw

Reply:
Yes, indeed. The control flow should know it's never undefined at that point. Maybe something for a bug report over at typescript. :)